Version incompatibility with `typing_extensions.deprecated`.

## Describe the Bug
The latest version imports `typing_extensions.deprecated`, which was added in [v4.5](https://typing-extensions.readthedocs.io/en/latest/index.html#deprecated). Pyproject.toml only requires version >=4.1.

```python
.venv/lib/python3.11/site-packages/strawberry/types/types.py:16: in <module>
    from typing_extensions import Self, deprecated
E   ImportError: cannot import name 'deprecated' from 'typing_extensions' 
```

## System Information

 - Operating system: any
 - Strawberry version (if applicable): 0.187.0

--- a/RELEASE.md
+++ b/RELEASE.md
@@ -1,0 +1,4 @@
+Release type: patch
+
+This release bumps the minimum requirement of
+`typing-extensions` to 4.5

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -36,7 +36,7 @@ build-backend = "poetry.masonry.api"
 [tool.poetry.dependencies]
 python = "^3.7"
 graphql-core = "~3.2.0"
-typing_extensions = ">=4.1.0"
+typing-extensions = ">=4.5.0"
 python-dateutil = "^2.7.0"
 starlette = {version = ">=0.18.0", optional = true}
 typer = {version = ">=0.7.0", optional = true}

--- a/tests/pyright/test_enum.py
+++ b/tests/pyright/test_enum.py
@@ -25,7 +25,7 @@ def test_enum_with_decorator():
     assert results == [
         Result(
             type="information",
-            message='Type of "IceCreamFlavour" is "Type[IceCreamFlavour]"',
+            message='Type of "IceCreamFlavour" is "type[IceCreamFlavour]"',
             line=12,
             column=13,
         ),
@@ -63,7 +63,7 @@ def test_enum_with_decorator_and_name():
     assert results == [
         Result(
             type="information",
-            message='Type of "Flavour" is "Type[Flavour]"',
+            message='Type of "Flavour" is "type[Flavour]"',
             line=12,
             column=13,
         ),
@@ -99,7 +99,7 @@ def test_enum_with_manual_decorator():
             type="information",
             message=(
                 'Type of "strawberry.enum(IceCreamFlavour)" '
-                'is "Type[IceCreamFlavour]"'
+                'is "type[IceCreamFlavour]"'
             ),
             line=11,
             column=13,
@@ -139,7 +139,7 @@ def test_enum_with_manual_decorator_and_name():
             type="information",
             message=(
                 'Type of "strawberry.enum(name="IceCreamFlavour")(Flavour)" '
-                'is "Type[Flavour]"'
+                'is "type[Flavour]"'
             ),
             line=11,
             column=13,
@@ -180,7 +180,7 @@ def test_enum_deprecated():
     assert results == [
         Result(
             type="information",
-            message='Type of "IceCreamFlavour" is "Type[IceCreamFlavour]"',
+            message='Type of "IceCreamFlavour" is "type[IceCreamFlavour]"',
             line=14,
             column=13,
         ),

--- a/tests/pyright/test_federation.py
+++ b/tests/pyright/test_federation.py
@@ -43,7 +43,7 @@ def test_federation_type():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=18,
             column=13,
         ),
@@ -92,7 +92,7 @@ def test_federation_interface():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=14,
             column=13,
         ),
@@ -139,7 +139,7 @@ def test_federation_input():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=12,
             column=13,
         ),

--- a/tests/pyright/test_federation_fields.py
+++ b/tests/pyright/test_federation_fields.py
@@ -69,7 +69,7 @@ def test_pyright():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=29,
             column=13,
         ),
@@ -82,7 +82,7 @@ def test_pyright():
         ),
         Result(
             type="information",
-            message='Type of "UserInput" is "Type[UserInput]"',
+            message='Type of "UserInput" is "type[UserInput]"',
             line=32,
             column=13,
         ),

--- a/tests/pyright/test_fields.py
+++ b/tests/pyright/test_fields.py
@@ -38,7 +38,7 @@ def test_pyright():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=13,
             column=13,
         ),

--- a/tests/pyright/test_fields_input.py
+++ b/tests/pyright/test_fields_input.py
@@ -37,7 +37,7 @@ def test_pyright():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=13,
             column=13,
         ),

--- a/tests/pyright/test_fields_resolver.py
+++ b/tests/pyright/test_fields_resolver.py
@@ -41,7 +41,7 @@ def test_pyright():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=17,
             column=13,
         ),

--- a/tests/pyright/test_fields_resolver_async.py
+++ b/tests/pyright/test_fields_resolver_async.py
@@ -41,7 +41,7 @@ def test_pyright():
         ),
         Result(
             type="information",
-            message='Type of "User" is "Type[User]"',
+            message='Type of "User" is "type[User]"',
             line=17,
             column=13,
         ),

--- a/tests/pyright/test_interface.py
+++ b/tests/pyright/test_interface.py
@@ -20,7 +20,7 @@ def test_pyright():
     assert results == [
         Result(
             type="information",
-            message='Type of "Node" is "Type[Node]"',
+            message='Type of "Node" is "type[Node]"',
             line=9,
             column=13,
         ),
@@ -45,7 +45,7 @@ def test_pyright_calling():
     assert results == [
         Result(
             type="information",
-            message='Type of "Node" is "Type[Node]"',
+            message='Type of "Node" is "type[Node]"',
             line=9,
             column=13,
         ),

--- a/tests/pyright/test_scalars.py
+++ b/tests/pyright/test_scalars.py
@@ -99,7 +99,7 @@ def test_schema_overrides():
     assert results == [
         Result(
             type="information",
-            message='Type of "EpochDateTime" is "Type[datetime]"',
+            message='Type of "EpochDateTime" is "type[datetime]"',
             line=17,
             column=13,
         ),

